flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/publisher/fanout/StreamConsumerRegistrar.java

```diff
@@ -243,28 +244,28 @@ private void waitForConsumerToDeregister(
     }
 
     private Optional<DescribeStreamConsumerResponse> describeStreamConsumer(
-            final String streamArn, final String streamConsumerName)
-            throws InterruptedException, ExecutionException {
+            final String streamArn, final String streamConsumerName) throws Exception {
         return describeStreamConsumer(
                 () ->
                         kinesisProxyV2Interface.describeStreamConsumer(
                                 streamArn, streamConsumerName));
     }
 
     private Optional<DescribeStreamConsumerResponse> describeStreamConsumer(
-            final String streamConsumerArn) throws InterruptedException, ExecutionException {
+            final String streamConsumerArn) throws Exception {
         return describeStreamConsumer(
                 () -> kinesisProxyV2Interface.describeStreamConsumer(streamConsumerArn));
     }
 
     private Optional<DescribeStreamConsumerResponse> describeStreamConsumer(
-            final ResponseSupplier<DescribeStreamConsumerResponse> responseSupplier)
-            throws InterruptedException, ExecutionException {
+            final Callable<DescribeStreamConsumerResponse> responseSupplier) throws Exception {
         DescribeStreamConsumerResponse response;
 
         try {
-            response = responseSupplier.get();
-        } catch (ExecutionException ex) {
+            response = responseSupplier.call();
+        } catch (Exception ex) {
+            LOG.warn("describeStreamConsumer caught ExecutionException: {}", ex);
```

Comment:
`caught ExecutionException` not necessarily now you are catching `Exception`. This change is slightly worrying, what happens now if an `InterruptedException` is thrown?